### ISSUE TITLE
Viper throws exception on math operations between uints

### ISSUE THREAD

**Issue by peterhinch** [micropython/micropython#6177]
The following fails with **ViperTypeError: can't do binary op between 'uint' and 'uint'**
```python
@micropython.viper
def foo(x:uint) -> uint:
    return x + uint(9)  # Every operator I tried failed here
```
Tested on Pyboard D SF6W. Ref [this forum thread](https://forum.micropython.org/viewtopic.php?f=2&t=8574).

**Comment by dpgeorge** [comment]
This looks like a duplicate of #1847.  I guess it's time to implement this feature.

**Comment by dpgeorge** [comment]
See #6178 for a fix.

**Comment by dpgeorge** [comment]
Implemented in 41fa8b5482089bdd7fa5478fe24f32913b23967c
